include/exiv2/error.hpp

```diff
@@ -253,6 +253,7 @@ namespace Exiv2 {
         kerCorruptedMetadata,
         kerArithmeticOverflow,
         kerMallocFailed,
+        kerNotAHeif,
```

Comment:
I noticed that you copied the style from the value `kerNotAJpeg`. IMHO I would prefer to refactor a bit here and have an unique value indicating "not expected format" rather than having  a new value for each image format. Opinions @clanmills @D4N ?

Reply:
> IMHO I would prefer to refactor a bit here and have an unique value indicating "not expected format" rather than having a new value for each image format.

Definitely agree on this one.